tests/by-util/test_cksum.rs

```diff
@@ -35,14 +35,19 @@ fn test_empty() {
 }
 
 #[test]
-#[ignore]
 fn test_arg_overrides_stdin() {
     let (at, mut ucmd) = at_and_ucmd!();
     let input = "foobarfoobar";
 
     at.touch("a");
 
-    let result = ucmd.arg("a").pipe_in(input.as_bytes()).run();
+    let result = ucmd
+        .arg("a")
+        .pipe_in(input.as_bytes())
+        // the command might have exited before all bytes have been pipe in.
```

Comment:
hmm, feels like a workaround
I thought you would fix the actual defect.

We should not a broken pipe, right?

Reply:
I agree it feels like a workaround, but if I understand the issue correctly, then I don't think we have many other options.

Here's what I think is going on:
- We spawn `cksum`
- We pipe `ipnut` into it
- By the time the `input` is actually fed `cksum`, it has already exited
- We get a `broken pipe`

So my ideas would be:
- `mksum` should consume stdin even if it doesn't need it
- not being able to feed stdin to `cksum` is okay in this case

Maybe I'm missing something, but 2) seems better because it only makes the test ugly while 1) would make `cksum` ugly.

If I should explore other options, I'd be happy to, but I'd need some hints :)

Reply:
Or in other words: This test checks what happens if `cksum` doesn't care about stdin and the result is "Broken pipe". I think that's kind of expected -- or at least not relevant. The test itself still checks whether the end result is correct (that is, the checksum of the provided file).

Reply:
OK, could you please open a new issue on cksum so that we keep track of this issue? Thanks

Reply:
I think there might be a misunderstanding.

As far as I understand things, this is not an issue with `cksum`. What we're testing here is "make sure cksum doesn't use stdin if a file is provided". And we do this by providing some data via stdin.

Whether the data we feed to stdin ever reaches `cksum` is rather coincidental and depends on how fast `cksum` is done with its actual work (which does not depend on stdin). In 99% of the cases, `cksum` does not exit before the test can pipe some data to it, but in the other 1% `cksum` is already done.

If `cksum` is already done with its actual work before we even got to feed data to it, we get a `Broken pipe`. But that's okay, because we're feeding data to a short-lived program that does not expect (and therefore await) any data from us via stdin.

Hope this clarifies my view. If you see it differently -- or suspect something else is wrong -- I'd be happy to explore that route :)